data_engineering.qmd

```diff
@@ -1,121 +1,526 @@
+|                                                                       |
+| You are receiving this message because one or more account holders    |
+| have requested that their voice clips be deleted. Their clips are     |
+| part of the dataset that you downloaded and are associated with the   |
+| hashed IDs listed below. Please delete them from your downloads in    |
+| order to fulfill your third party data privacy obligations.           |
+|                                                                       |
+| Thank you for your timely completion.                                 |
+|                                                                       |
+| -   4497f1df0c6c4e647fa4354ad07a40075cc95a210dafce49ce0c35cd252       |
+| e4ec0fad1034e0cc3af869499e6f60ce315fe600ee2e9188722de906f909a21e0ee57 |
+|                                                                       |
+| -   97a8f0a1df086bd5f76343f5f4a511ae39ec98256a0ca48de5c54bc5771       |
+| d8c8e32283a11056147624903e9a3ac93416524f19ce0f9789ce7eef2262785cf3af7 |
+|                                                                       |
+| -   969ea94ac5e20bdd7a098747f5dc2f6d203f6b659c0c3b6257dc790dc34       |
+| d27ac3f2fafb3910f1ec8d7ebea38c120d4b51688047e352baa957cc35f0f5c69b112 |
+|                                                                       |
+| -   6b5460779f644ad39deffeab6edf939547f206596089d554984abff3d36       |
+| a4ecc06e66870958e62299221c09af8cd82864c626708371d72297eaea5955d8e46a9 |
+|                                                                       |
+| -   33275ff207a27708bd1187ff950888da592cac507e01e922c4b9a07d3f6       |
+| c2c3fe2ade429958c3702294f446bfbad8c4ebfefebc9e157d358ccc6fcf5275e7564 |
++=======================================================================+
++-----------------------------------------------------------------------+
+
+Having the ability to update the dataset by removing data from the dataset will enable the dataset creators to uphold legal and ethical obligations around data usage and privacy. However, the ability to remove data has some important limitations. We need to think about the fact that some models may have already been trained on the dataset and there is no clear or known way to eliminate a particular data sample\'s effect from the trained network. There is no erase mechanism. Thus, this begs the question, should the model be re-trained from scratch each time a sample is removed? That\'s a costly option. Once data has been used to train a model, simply removing it from the original dataset may not fully eliminate[^8]^,^[^9]^,^[^10] its impact on the model\'s behavior. New research is needed around the effects of data removal on already-trained models and whether full retraining is necessary to avoid retaining artifacts of deleted data. This presents an important consideration when balancing data licensing obligations with efficiency and practicality in an evolving, deployed ML system.
```

Comment:
This is a really interesting! The references noted are very helpful in exploring this topic, but I think it would be great to use the keyword "machine unlearning" in this section. This is an emerging field with many recent works proposing alternative methods to full retraining! An interesting note is that while companies may retrain/alter their models to comply with these regulations, if companies at any point allow users to directly download the weights of a model there will never be any way to completely erase data.

Reply:
+1. Some Citations. The first introduces the subject, the second has a more detailed exploration of some of the risks ('leakage') of even having access to just gradients of a model.

https://ieeexplore.ieee.org/abstract/document/9519428?casa_token=4pvI0EGYJqcAAAAA:cn3JnC5O1cgV6Z29Aaoex_kaJIghLFtLmDvhaOsL_4-wFeXHmznIVXBzXFeW_xC6JBIx-v9Q

https://arxiv.org/abs/1911.03030